File: Round C/Countdown/Ghiurcuta-Cpp/Countdown.cpp
#include <iostream>

using namespace std;

int main()
{
	int t;
	cin >> t;
	int n, k;
	for (int i = 0; i < t; i++)
	{
		cin >> n >> k;
		int ans = 0;
		int arr[n];

		for (int j = 0; j < n; j++)
		{
			cin >> arr[j];
		}
		for (int j = 0; j < n; j++)
		{
			if (arr[j] == k)
			{
				int x = k;
				for (int a = j; a < j + k; a++)
				{
					if (arr[a] == x)
					{
						x--;
					}

					else
					{
						break;
					}
					
				}
				if (x == 0)
				{
					ans++;
				}
			}
		}
		cout << "Case #" << i + 1 << ": " << ans << endl;
	}
}

File: Round C/Countdown/Niranjan-Python/countdown.py
t = int(input())
for _ in range(t):
    n, k = map(int, input().split(" "))
    arr = [int(x) for x in input().split(" ")]
    ans = 0
    c = 0
    for i in range(1,n):
        if arr[i] == arr[i-1] -1:
            c += 1
        else:
            c = 0
        if arr[i] == 1 and c >= k - 1:
            ans += 1
    print('Case #{}: {}'.format(_+1,ans))